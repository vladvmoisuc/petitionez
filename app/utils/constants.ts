export const DATE_FORMATS = {
  STANDARD: "D/MM/YYYY",
};

export const ROUTES = {
  MANAGEMENT: "/management",
  CAMPAIGNS: "/campaigns",
  LOGIN: "/login",
};

export const EMAIL_TYPES = {
  NEW: "new",
  OLD: "old",
};

export const LOGIN_STATES = {
  ...EMAIL_TYPES,
  DEFAULT: "default",
  ERROR: "error",
};

export const INTENTS = {
  VALIDATE: "validate",
  GENERATE: "generate",
  DELETE: {
    CAMPAIGN: "delete-campaign",
  },
  UPDATE: {
    CAMPAIGN: "update-campaign",
  },
  CREATE: {
    ENTITY: "create-entity",
    CAMPAIGN: "create-campaign",
  },
};

export const SLUGS = {
  CREATE: "create",
};

export const LINKS = {
  POSTIMAGES: {
    URL: "https://postimages.org/",
    NAME: "postimages.org",
  },
};
