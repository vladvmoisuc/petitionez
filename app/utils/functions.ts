import ro from "~/content/ro.json";
import errors from "~/content/errors.json";
import logs from "~/content/logs.json";

import type { DynamicObject } from "~/utils/types";

export function waitFor(time: number = 500) {
  return new Promise<void>((resolve) =>
    setTimeout(() => {
      resolve();
    }, time)
  );
}

export function replace(data: { [key: string]: string }, content: string) {
  return Object.keys(data).reduce(
    (text, key) => text.replace(new RegExp(`{{${key}}}`, "g"), data[key]),
    content
  );
}

function extractContent(
  source: DynamicObject,
  key: string,
  values?: DynamicObject
): string {
  const content = source[key as keyof typeof source] ?? "";

  return values ? replace(values, content) : content;
}

export function getText(key: string, values?: DynamicObject): string {
  return extractContent(ro, key, values);
}

export function getError(key: string, values?: DynamicObject): string {
  return extractContent(errors, key, values);
}

export function getLog(key: string, values?: DynamicObject): string {
  return extractContent(logs, key, values);
}

export async function getFormData(
  request: Request
): Promise<{ keys: string[]; values: DynamicObject }> {
  const data = await request.formData();
  const keys = [...data.keys()];

  const values = keys.reduce((values, key) => {
    return {
      ...values,
      [key]: data.get(key),
    };
  }, {});

  return {
    keys,
    values,
  };
}
