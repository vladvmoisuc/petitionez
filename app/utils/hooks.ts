import { useState, useEffect, useCallback } from "react";

import type { Transition } from "@remix-run/react/dist/transition";

export const useLocalStorage = <T>(
  key: string
): [T | undefined, typeof setLocalStorage] => {
  const [state, setState] = useState();

  const isClientSide = typeof window !== "undefined";

  useEffect(() => {
    const value = window.localStorage.getItem(key);

    if (isClientSide && value) {
      setState(typeof value === "string" ? JSON.parse(value) : value);
    }
  }, [isClientSide, key]);

  const setLocalStorage = useCallback(
    (data: any) => {
      const value = typeof data !== "string" ? JSON.stringify(data) : data;

      if (isClientSide) {
        window.localStorage.setItem(key, value);

        setState({ ...data });
      }
    },
    [isClientSide, key]
  );

  return [state, setLocalStorage];
};

export const useAlert = (transition: Transition, time: number = 1000) => {
  const [alert, setAlert] = useState(false);

  useEffect(() => {
    if (transition.state === "loading" && transition.type === "actionReload") {
      setAlert(true);

      setTimeout(() => {
        setAlert(false);
      }, time);
    }
  }, [transition.state, transition.type, time]);

  return alert;
};
