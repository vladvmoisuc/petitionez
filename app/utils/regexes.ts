export default {
  EMAIL: /^\S+@\S+\.\S+$/,
  URL: /^(http:\/\/|https:\/\/).+\..+/,
  START_LETTER: /^[a-z]/g,
  DASHES: /-/g,
  SPACES: /\s/g,
};
