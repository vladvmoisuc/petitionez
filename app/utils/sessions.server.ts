import { createCookieSessionStorage } from "@remix-run/node";

const HOUR = 1 * 60 * 60; // second -> minute -> hour

const { getSession, commitSession, destroySession } =
  createCookieSessionStorage({
    cookie: {
      name: "__session",
      httpOnly: true,
      maxAge: HOUR,
      path: "/",
      sameSite: "lax",
      secrets: [process.env.COOKIES_SECRET_KEY as string],
      secure: true,
    },
  });

async function getLoggedUser(request: Request) {
  const session = await getSession(request.headers.get("Cookie"));

  const loggedUserEmail: string = session.get("email") ?? "";
  const loggedUserId: string = session.get("id") ?? "";
  const isAdmin: boolean = session.get("admin") ?? false;

  return { loggedUserEmail, loggedUserId, isAdmin };
}

export { getSession, commitSession, destroySession, getLoggedUser };
