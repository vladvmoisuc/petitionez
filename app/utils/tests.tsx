import { createMemoryRouter, RouterProvider } from "react-router-dom";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import type { ReactElement, ReactNode } from "react";

export function renderWithUser(jsx: ReactElement) {
  return {
    user: userEvent.setup({ delay: null }),
    ...render(jsx),
  };
}

export function renderWithRouter(jsx: ReactElement) {
  const Wrapper = ({ children }: { children: ReactNode }) => (
    <RouterProvider
      router={createMemoryRouter([
        {
          path: "/",
          element: children,
        },
      ])}
    />
  );

  return {
    user: userEvent.setup({ delay: null }),
    ...render(jsx, {
      wrapper: Wrapper,
    }),
  };
}
