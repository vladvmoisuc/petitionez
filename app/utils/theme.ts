import { createTheme, colors } from "@mui/material";

const FONTS = {
  PLAYFAIR_DISPLAY: "'Playfair Display', serif",
  RED_HAT: "'Red Hat Mono', monospace",
};

export default createTheme({
  typography: {
    h1: {
      fontFamily: FONTS.PLAYFAIR_DISPLAY,
    },
    h2: {
      fontFamily: FONTS.PLAYFAIR_DISPLAY,
    },
    h3: {
      fontFamily: FONTS.PLAYFAIR_DISPLAY,
    },
    h4: {
      fontFamily: FONTS.PLAYFAIR_DISPLAY,
    },
    h5: {
      fontFamily: FONTS.PLAYFAIR_DISPLAY,
    },
    fontFamily: FONTS.RED_HAT,
  },
  components: {
    MuiAppBar: {
      defaultProps: {
        color: "transparent",
        elevation: 0,
        position: "static",
      },
    },
    MuiToolbar: {
      styleOverrides: {
        root: {
          justifyContent: "space-between",
          borderBottomWidth: 1,
          borderBottomStyle: "solid",
          borderBottomColor: colors.grey[500],
        },
      },
    },
    MuiCard: {
      defaultProps: {
        elevation: 0,
        square: true,
      },
    },
    MuiSkeleton: {
      styleOverrides: {
        root: {
          backgroundColor: colors.grey[100],
        },
      },
    },
    MuiContainer: {
      defaultProps: {
        maxWidth: false,
      },
    },
  },
});
