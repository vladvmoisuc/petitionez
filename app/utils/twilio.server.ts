import sendgrid from "@sendgrid/mail";
import { readFileSync } from "fs";
import { join } from "path";

import { replace } from "~/utils/functions";

sendgrid.setApiKey(process.env.SENDGRID_API_KEY as string);

const TEMPLATES = {
  OTP: {
    from: "petitionez@gmail.com",
    subject: "Cod de autentificare petitionez.ro",
    html: readFileSync(
      join(__dirname, "..", "app", "templates", "emails", "otp.html")
    ).toString(),
  },
};

type Params = {
  type: keyof typeof TEMPLATES;
  to: string;
  data: {
    [key: string]: string;
  };
};

export default {
  send: async ({ type, to, data = {} }: Params) =>
    await sendgrid.send({
      ...TEMPLATES[type],
      to,
      html: replace(data, TEMPLATES[type].html),
    }),
};
