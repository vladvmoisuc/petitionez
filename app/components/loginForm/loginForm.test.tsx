import { screen } from "@testing-library/react";

import LoginForm from ".";

import { renderWithRouter } from "~/utils/tests";
import { getText } from "~/utils/functions";

import { LOGIN_STATES } from "~/utils/constants";

const props = {
  code: "",
  email: "",
  handleCodeChange: jest.fn(),
  handleEmailChange: jest.fn(),
};

describe("<LoginForm />", () => {
  it("should match the snapshot", () => {
    const { container } = renderWithRouter(<LoginForm {...props} />);

    expect(container).toMatchSnapshot();
  });

  it(`should match the snapshot when the type is set to ${LOGIN_STATES.ERROR}`, () => {
    const { container } = renderWithRouter(
      <LoginForm
        {...props}
        type={LOGIN_STATES.ERROR as keyof typeof LOGIN_STATES}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it(`should match the snapshot when the type is set to ${LOGIN_STATES.NEW}`, () => {
    const { container } = renderWithRouter(
      <LoginForm
        {...props}
        type={LOGIN_STATES.NEW as keyof typeof LOGIN_STATES}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it(`should match the snapshot when the type is set to ${LOGIN_STATES.OLD}`, () => {
    const { container } = renderWithRouter(
      <LoginForm
        {...props}
        type={LOGIN_STATES.OLD as keyof typeof LOGIN_STATES}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it(`should call the handleCodeChange prop when the code is changed`, async () => {
    const { container, user } = renderWithRouter(
      <LoginForm
        {...props}
        type={LOGIN_STATES.NEW as keyof typeof LOGIN_STATES}
      />
    );

    await user.type(screen.getByLabelText(getText("login.fields.code")), "1");

    expect(props.handleCodeChange).toBeCalled();

    expect(container).toMatchSnapshot();
  });

  it(`should call the handleEmailChange prop when the email is changed`, async () => {
    const { container, user } = renderWithRouter(<LoginForm {...props} />);

    await user.type(
      screen.getByPlaceholderText(getText("login.fields.email")),
      "j"
    );

    expect(props.handleEmailChange).toBeCalled();

    expect(container).toMatchSnapshot();
  });
});
