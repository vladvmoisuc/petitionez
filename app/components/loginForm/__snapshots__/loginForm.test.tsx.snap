// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should call the handleCodeChange prop when the code is changed 1`] = `
<div>
  <form
    action="/"
    class="horizontally-centered full-width"
    method="post"
  >
    <div
      class="css-1wz31qk-MuiStack-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
      >
        Cod trimis pe email
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-11cn0eh-MuiTypography-root"
      >
        Tocmai ți-am trimis un email cu codul de autenficare.
      </h6>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-readOnly css-q0jhri-MuiInputBase-root-MuiInput-root"
      >
        <input
          class="MuiInputBase-input MuiInput-input MuiInputBase-readOnly css-1x51dt5-MuiInputBase-input-MuiInput-input"
          name="email"
          placeholder="Email"
          readonly=""
          type="text"
          value=""
        />
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r2:"
          id=":r2:-label"
        >
          Cod
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id=":r2:"
            name="code"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        name="intent"
        tabindex="-1"
        type="submit"
        value="validate"
      >
        Validează codul
      </button>
    </div>
  </form>
</div>
`;

exports[`<LoginForm /> should call the handleEmailChange prop when the email is changed 1`] = `
<div>
  <form
    action="/"
    class="horizontally-centered full-width"
    method="post"
  >
    <div
      class="css-1wz31qk-MuiStack-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
      >
        Logare
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-11cn0eh-MuiTypography-root"
      >
        Vei primi un cod unic pe adresa de email. Folosește acel cod pentru a finaliza autentificarea.
      </h6>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-focused css-q0jhri-MuiInputBase-root-MuiInput-root"
      >
        <input
          class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          name="email"
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        name="intent"
        tabindex="-1"
        type="submit"
        value="generate"
      >
        Loghează-te în aplicație
      </button>
    </div>
  </form>
</div>
`;

exports[`<LoginForm /> should match the snapshot 1`] = `
<div>
  <form
    action="/"
    class="horizontally-centered full-width"
    method="post"
  >
    <div
      class="css-1wz31qk-MuiStack-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
      >
        Logare
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-11cn0eh-MuiTypography-root"
      >
        Vei primi un cod unic pe adresa de email. Folosește acel cod pentru a finaliza autentificarea.
      </h6>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-q0jhri-MuiInputBase-root-MuiInput-root"
      >
        <input
          class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          name="email"
          placeholder="Email"
          type="text"
          value=""
        />
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        name="intent"
        tabindex="-1"
        type="submit"
        value="generate"
      >
        Loghează-te în aplicație
      </button>
    </div>
  </form>
</div>
`;

exports[`<LoginForm /> should match the snapshot when the type is set to error 1`] = `
<div>
  <form
    action="/"
    class="horizontally-centered full-width"
    method="post"
  >
    <div
      class="css-1wz31qk-MuiStack-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
      >
        Ceva nu a funcționat
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-11cn0eh-MuiTypography-root"
      >
        A apărut o problemă. E posibil ca unul dintre servicii să nu funcționeze, adresa introdusă să nu existe în baza noastră de date sau codul să nu fie scris corect.
      </h6>
      <a
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        href="/login"
        tabindex="0"
      >
        Încearcă din nou
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </form>
</div>
`;

exports[`<LoginForm /> should match the snapshot when the type is set to new 1`] = `
<div>
  <form
    action="/"
    class="horizontally-centered full-width"
    method="post"
  >
    <div
      class="css-1wz31qk-MuiStack-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
      >
        Cod trimis pe email
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-11cn0eh-MuiTypography-root"
      >
        Tocmai ți-am trimis un email cu codul de autenficare.
      </h6>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-readOnly css-q0jhri-MuiInputBase-root-MuiInput-root"
      >
        <input
          class="MuiInputBase-input MuiInput-input MuiInputBase-readOnly css-1x51dt5-MuiInputBase-input-MuiInput-input"
          name="email"
          placeholder="Email"
          readonly=""
          type="text"
          value=""
        />
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Cod
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id=":r0:"
            name="code"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        name="intent"
        tabindex="-1"
        type="submit"
        value="validate"
      >
        Validează codul
      </button>
    </div>
  </form>
</div>
`;

exports[`<LoginForm /> should match the snapshot when the type is set to old 1`] = `
<div>
  <form
    action="/"
    class="horizontally-centered full-width"
    method="post"
  >
    <div
      class="css-1wz31qk-MuiStack-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
      >
        Cod trimis anterior pe email
      </h1>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-11cn0eh-MuiTypography-root"
      >
        Un email cu codul de autenficare ți-a fost trimis în ultimele 30 de minute.
      </h6>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-readOnly css-q0jhri-MuiInputBase-root-MuiInput-root"
      >
        <input
          class="MuiInputBase-input MuiInput-input MuiInputBase-readOnly css-1x51dt5-MuiInputBase-input-MuiInput-input"
          name="email"
          placeholder="Email"
          readonly=""
          type="text"
          value=""
        />
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          Cod
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id=":r1:"
            name="code"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        name="intent"
        tabindex="-1"
        type="submit"
        value="validate"
      >
        Validează codul
      </button>
    </div>
  </form>
</div>
`;
